### PR TITLE
#17994 : As of many versions ago, pushing an archived content would m…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…ake dotCMS delete it altogether in the receiving instance, which is incorrect. This change allows the Push Publishing feature and the Push Now actionlet the flexibility to push an archived content to the receiver for: (1) Simply flagging it as Archived, or (2) to delete it completely.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 1]
```diff
@@ -14,122 +18,135 @@
...
+    private static final String ACTIONLET_DESCRIPTION = "This actionlet will automatically publish the the content " +
+            "object to the specified environment(s). Multiple environments can be separated by a comma (',')";
+    private static final String PARAM_ENVIRONMENT = "environment";
+    private static final String PARAM_FORCE_PUSH = "force";
+
     private PublisherAPI publisherAPI = PublisherAPI.getInstance();
+    private EnvironmentAPI environmentAPI = APILocator.getEnvironmentAPI();
+    private BundleAPI bundleAPI = APILocator.getBundleAPI();
+    private UserAPI userAPI = APILocator.getUserAPI();
+    private RoleAPI roleAPI = APILocator.getRoleAPI();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Private field 'roleAPI' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 2]
```diff
@@ -14,122 +18,135 @@
...
-
-            Contentlet ref = processor.getContentlet();
-
-            boolean _contentPushNeverExpire = true;
-
-            boolean forcePush = ("true".equals(params.get("force").getValue())) ? true : false;
-            String envrions = params.get("environment").getValue();
-            if (envrions == null) {
-                Logger.error(this.getClass(), "There are no environments set to push to");
+            final boolean forcePush = "true".equals(params.get(PARAM_FORCE_PUSH).getValue()) ? Boolean.TRUE : Boolean.FALSE;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 3]
```diff
@@ -14,122 +18,135 @@
...
+
+    private static final String ENVIRONMENT_DELIMITER = ",";
+    private static final String ACTIONLET_NAME = "Push Now";
+    private static final String ACTIONLET_DESCRIPTION = "This actionlet will automatically publish the the content " +
+            "object to the specified environment(s). Multiple environments can be separated by a comma (',')";
+    private static final String PARAM_ENVIRONMENT = "environment";
+    private static final String PARAM_FORCE_PUSH = "force";
+
     private PublisherAPI publisherAPI = PublisherAPI.getInstance();
+    private EnvironmentAPI environmentAPI = APILocator.getEnvironmentAPI();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Private field 'environmentAPI' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 4]
```diff
@@ -14,122 +18,135 @@
...
+    private BundleAPI bundleAPI = APILocator.getBundleAPI();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Private field 'bundleAPI' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 5]
```diff
@@ -14,122 +18,135 @@
...
+    private UserAPI userAPI = APILocator.getUserAPI();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Private field 'userAPI' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 6]
```diff
@@ -14,122 +18,135 @@
...
+                    permissionedEnv.add(environment);
+                }
+            } else {
+                for (final Role role : roleList){
                     try {
-                        permissionedEnv.addAll(APILocator.getEnvironmentAPI().findEnvironmentsByRole(r.getId()));
-                    } catch (Exception e) {
-                        Logger.error(PushNowActionlet.class, e.getMessage());
+                        permissionedEnv.addAll(this.environmentAPI.findEnvironmentsByRole(role.getId()));
+                    } catch (final Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java [hunk 7]
```diff
@@ -14,122 +18,135 @@
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5077051)

**Review by nollymar — APPROVED** [review]

**Comment by erickgonzalez** [comment]
Since we are doing some changes regarding PP and all its dialogs(changing to angular dialogs), I added this issue to the EPIC. So I think this should not be merged since it will be attacked in Angular.

**Review by erickgonzalez — CHANGES_REQUESTED** [review]
Since we are doing some changes regarding PP and all its dialogs(changing to angular dialogs), I added this issue to the EPIC. So I think this should not be merged since it will be attacked in Angular.

**Review by dsilvam — APPROVED** [review]

**Pull request merged**